### PR TITLE
Added cmdlets and tests for NetworkWatcher APIs

### PR DESCRIPTION
**Pull request by irrogozh**
## Description

Fixes based on CR, resolved static analysis issues
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@irrogozh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1399/

File: src/ResourceManager/Network/Network.sln [hunk 1]
```diff
@@ -33,61 +35,121 @@ EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
+		Debug|x64 = Debug|x64
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Network.sln]: @irrogozh please remove all of the below references to x64 build

File: src/ResourceManager/Common/Commands.ResourceManager.Common/Commands.ResourceManager.Common.csproj [hunk 2]
```diff
@@ -65,6 +65,10 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Azure.Common.2.1.0\lib\net45\Microsoft.Azure.Common.NetFramework.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=8.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/Common/Commands.ResourceManager.Common/Commands.ResourceManager.Common.csproj]: @irrogozh please remove this reference to Network from ResourceManager

File: src/ResourceManager/Common/Commands.ResourceManager.Common/packages.config [hunk 3]
```diff
@@ -5,6 +5,7 @@
   <package id="Microsoft.AspNet.WebApi.Client" version="5.2.2" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Network" version="8.3.0-preview" targetFramework="net45" />
```
Comment (cormacpayne) [inline src/ResourceManager/Common/Commands.ResourceManager.Common/packages.config]: @irrogozh same comment

File: src/ResourceManager/Common/Commands.ResourceManager.Common/Commands.ResourceManager.Common.csproj (hunk 2 above)
Comment (irrogozh) [inline src/ResourceManager/Common/Commands.ResourceManager.Common/Commands.ResourceManager.Common.csproj]: Done

File: src/ResourceManager/Common/Commands.ResourceManager.Common/packages.config (hunk 3 above)
Comment (irrogozh) [inline src/ResourceManager/Common/Commands.ResourceManager.Common/packages.config]: Done

File: src/ResourceManager/Network/Network.sln (hunk 1 above)
Comment (irrogozh) [inline src/ResourceManager/Network/Network.sln]: Done

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Comment by cormacpayne** [comment]
On-demand build as passed: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1408/

Just restarted build for this PR since it was previously run on a different VM that was not configured

LGTM once build passes

**Pull request merged**